File: utils/dev_utils/supabase_table_admin.py
"""supabase admin"""

import os
from supabase import create_client, Client

# Initialize Supabase client
url: str = os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
key: str = os.environ.get("NEXT_PUBLIC_SUPABASE_SERVICE_ROLE_KEY")
supabase: Client = create_client(url, key)


def setup_user_usage_table():
    """Create user_usage and user_profile tables"""


if __name__ == "__main__":
    setup_user_usage_table()

File: utils/dev_utils/supabase_table_admin.ts
import { createClient, SupabaseClient } from '@supabase/supabase-js'

// Initialize Supabase client with type safety
const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL as string
const supabaseServiceKey = process.env.NEXT_PUBLIC_SUPABASE_SERVICE_ROLE_KEY as string

if (!supabaseUrl || !supabaseServiceKey) {
  throw new Error('Missing Supabase environment variables')
}

const supabase: SupabaseClient = createClient(supabaseUrl, supabaseServiceKey)

async function setupUserUsageTable(): Promise<void> {
  // Function implementation will go here
  // You can add table creation logic using supabase.rpc() or raw SQL queries
}

// For development/CLI usage
if (require.main === module) {
  setupUserUsageTable()
    .catch((error) => {
      console.error('Error setting up tables:', error)
      process.exit(1)
    })
}

export { supabase, setupUserUsageTable }
